### PR TITLE
Added delete button in slideshow

### PR DESCRIPTION
**Pull request by manishbisht**
Fixes: #649 

Licence:  AGPL
### Reviewers

@oparoz @tempico

### CODE REVIEW

**Comment by tempico** [comment]
http://stackoverflow.com/questions/1580596/how-do-i-make-git-ignore-file-mode-chmod-changes

**Comment by oparoz** [comment]
Ah, I see that it's now down to a single commit :)
First time I see all those chmods though...

Thanks for the link @tempico

**Comment by manishbisht** [comment]
Good link @tempico I doesn't know about this earlier.
Thanks for your help ;)

**Comment by manishbisht** [comment]
Any Update on PR @oparoz ?

File: js/slideshowcontrols.js [hunk 1]
```diff
@@ -417,6 +420,95 @@
 		_setName: function (imageName) {
 			var nameElement = this.container.find('.title');
 			nameElement.text(imageName);
+		},
+
+		do_delete:function(files, dir) {
```
Comment (oparoz) [inline js/slideshowcontrols.js]: Why is this method here? It doesn't do anything useful afaics

File: templates/part.content.php [hunk 2]
```diff
@@ -118,6 +118,12 @@
 			<div class="album-info-loader"></div>
 			<div class="album-info-content markdown-body"></div>
 		</div>
+		<!-- button for opening bin -->
```
Comment (oparoz) [inline templates/part.content.php]: This button should be part of a separate PR. We don't have enough room to add a new button atm. It will be easier once buttons can be shown on 2-3 rows.

File: js/slideshowcontrols.js [hunk 3]
```diff
@@ -417,6 +420,95 @@
...
+		 * @private
+		 */
+		_deleteImage: function () {
+			var imageName = this.images[this.current].name;
+			var imagePath = this.images[this.current].path;
+			var fileId = Gallery.imageMap[imagePath].fileId;
+			var self=this;
+			$.ajax({
+				type:'DELETE',
+				url:OC.getRootPath()+'/remote.php/webdav/'+imagePath,
```
Comment (oparoz) [inline js/slideshowcontrols.js]: Replace with
https://github.com/owncloud/gallery/blob/0f4df90a064bee39d7330cd04fcf906ab23417fb/js/galleryutility.js#L120

File: js/slideshowcontrols.js [hunk 4]
```diff
@@ -417,6 +420,95 @@
...
+		},
+		/**
+		 * Delete the image from the slideshow
+		 * @private
+		 */
+		_deleteImage: function () {
+			var imageName = this.images[this.current].name;
+			var imagePath = this.images[this.current].path;
+			var fileId = Gallery.imageMap[imagePath].fileId;
+			var self=this;
```
Comment (oparoz) [inline js/slideshowcontrols.js]: Formatting. It's always a good idea to ask the IDE to format your files before making the commit.

File: js/slideshowcontrols.js [hunk 5]
```diff
@@ -417,6 +420,95 @@
...
+				url:OC.getRootPath()+'/remote.php/webdav/'+imagePath,
+				success: function (){
+					self.images.splice(self.current, 1);
+					delete self.images[self.current];
+					delete Gallery.imageMap[imagePath];
+					fileId = fileId.substring(1, fileId.length-1);
+					delete Thumbnails.map[fileId];
+					Gallery.albumMap[Gallery.currentAlbum].images.splice(self.current, 1);
+					if(self.images.length == 0)
+					{
```
Comment (oparoz) [inline js/slideshowcontrols.js]: Formatting

**Comment by oparoz** [comment]
Apart from my comments, you still need to fix the file permissions issue. Then we'll be able to test things and look at why the refresh doesn't work.

**Comment by oparoz** [comment]
Overall, you got the idea. You just need to get a bit more familiar to the way git works and pay more attention to details (reuse of existing functions, formatting, dead code).

**Comment by manishbisht** [comment]
@oparoz I will make the suggested changes. Can I get the more details about the file permissions issue.
